File: Algo/DynamicProgramming/TakeYouForward/practice/tuf/Sorting/InsertionSort.java
package tuf.Sorting;

import java.util.Arrays;
import java.util.stream.Collectors;

public class InsertionSort {

    int[] arr ={4,2,0,1,23,45234,-12,23343};

    public static void main(String[] args) {
        new InsertionSort();
    }

     InsertionSort(){
        sort(arr);
        print(arr);
    }

     public void sort(int[] arr){
        int N= arr.length;
        for(int i=0;i<N;i++){
            
            for(int j=i;j>0;j--){
                if(arr[j]>arr[j-1]) swap(arr,j,j-1);
            }
            
        }
    }

    public void swap(int[] arr,int src,int dest){
        int temp = arr[src];
        arr[src]= arr[dest];
        arr[dest]=temp;

}

    public void print(int[] arr){
        System.out.println(Arrays.stream(arr).boxed().collect(Collectors.toList()));
    }
}

File: Algo/DynamicProgramming/TakeYouForward/practice/practice3/Sorting/insertionsort.go
package main

import "fmt"

func swap(src int, dest int, arr [8]int) [8]int {
	test := arr[src]
	arr[src] = arr[dest]
	arr[dest] = test
	return arr
}

func main() {
	fmt.Print("Insertion Sort")
	arr := [8]int{6, 5, 7, 4, 8, 3, 9, 1}
	n := len(arr)
	for i := 0; i < n-1; i++ {
		for j := i + 1; j > 0; j-- {
			if arr[j-1] < arr[j] {
				arr = swap(j-1, j, arr)
			}
		}
	}
	fmt.Print(arr)
}
